### PR TITLE
v2: add TailnetSettingsResource

### PR DESCRIPTION
**Pull request by oxtoacart**
Supports getting and updating Tailnet settings.

Updates tailscale/corp#21628

### CODE REVIEW

File: v2/tailnet_settings.go [hunk 1]
```diff
@@ -0,0 +1,73 @@
+package tsclient
+
+import (
+	"context"
+	"net/http"
+)
+
+// TailnetSettingsResource provides an API to view/control settings for a tailnet.
```
Comment (soniaappasamy) [inline v2/tailnet_settings.go:8]: I wonder if we should start linking to api docs from some of these comments

File: v2/tailnet_settings.go [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+		RegionalRoutingOn           *bool `json:"regionalRoutingOn,omitempty"`
+		PostureIdentityCollectionOn *bool `json:"postureIdentityCollectionOn,omitempty"`
+	}
+
+	// RoleAllowedToJoinExternalTailnets constrains which users are allowed to join external tailnets
+	// based on their role.
+	RoleAllowedToJoinExternalTailnets string
+)
+
+const (
```
Comment (soniaappasamy) [inline v2/tailnet_settings.go:48]: Nit: add docs on the below three consts

**Review by soniaappasamy — APPROVED** [review]

File: v2/tailnet_settings.go (hunk 1 above)
Comment (oxtoacart) [reply v2/tailnet_settings.go:8]: Seems worthwhile.

File: v2/tailnet_settings.go (hunk 2 above)
Comment (oxtoacart) [reply v2/tailnet_settings.go:49]: I've added links to the API docs and am letting those speak for themselves.

**Pull request merged**